### PR TITLE
Refactored UI abstraction and fixed messaging tests

### PR DESCRIPTION
**Pull request by honza-kasik**

### CODE REVIEW

**Comment by honza-kasik** [comment]
Hmm... the check has passed but it still looks like it is pending.

File: common/src/main/java/org/jboss/hal/testsuite/page/config/MessagingPage.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public ConfigFragment getWindowFragment() {
         return  Graphene.createPageFragment(ConfigFragment.class, editPanel);
     }
 
-    public void navigateToMessagingProvider() {
+    private FinderNavigation createNavigationToMessagingSubsystem() {
```
Comment (zebrik) [inline common/src/main/java/org/jboss/hal/testsuite/page/config/MessagingPage.java]: You could reuse `ConfigPage#getSubsystemNavigation`

File: common/src/main/java/org/jboss/hal/testsuite/page/config/MessagingPage.java [hunk 2]
```diff
@@ -58,58 +58,63 @@ public void navigateToMessagingProvider() {
...
+                .step(SETTINGS_LABEL, MESSAGING_PROVIDER_LABEL)
+                .step(MESSAGING_PROVIDER_LABEL, name)
+                .selectRow();
+       Console.withBrowser(browser).dismissReloadRequiredWindowIfPresent();
+       finderFragment.invoke(action);
     }
 
-    public void selectConnectionsView() {
-        selectView("Connections");
+    public void viewMessagingProvider(String name) {
```
Comment (zebrik) [inline common/src/main/java/org/jboss/hal/testsuite/page/config/MessagingPage.java]: I'm missing `Application.waitUntilVisible();` in this method but anyway it seems to never be used so please remove this method.

File: common/src/main/java/org/jboss/hal/testsuite/page/config/MessagingPage.java [hunk 3]
```diff
@@ -58,58 +58,63 @@ public void navigateToMessagingProvider() {
...
         }
-        navigation.step(FinderNames.SUBSYSTEM, MESSAGING_SUBSYSTEM_LABEL);
-        navigation.step(SETTINGS_LABEL, MESSAGING_PROVIDER_LABEL);
+        return navigation.step(FinderNames.SUBSYSTEM, MESSAGING_SUBSYSTEM_LABEL);
     }
 
-    public void selectProvider(String provider) {
-        navigateToMessagingProvider();
-        navigation.step(MESSAGING_PROVIDER_LABEL, provider);
+    public void navigate() {
```
Comment (zebrik) [inline common/src/main/java/org/jboss/hal/testsuite/page/config/MessagingPage.java]: Never used. Does nothing useful. Please remove.

**Review by zebrik — APPROVED** [review]
Just minor suggestions.

**Comment by honza-kasik** [comment]
Thanks! I worked in all suggestions!

**Comment by zebrik** [comment]
Thank you! Merging...

**Pull request merged**